### PR TITLE
First go at moving to xo

### PR DESCRIPTION
**Pull request by OmgImAlexis**
@p0psicles @Thraxis 

This uses https://github.com/sindresorhus/xo and https://github.com/sindresorhus/atom-linter-xo

- [x] PR is based on the DEVELOP branch
- [ ] Don't send big changes all at once. Split up big PRs into multiple smaller PRs that are easier to manage and review
- [x] Read [contribution guide](https://github.com/PyMedusa/SickRage/blob/master/.github/CONTRIBUTING.md)

### CODE REVIEW

File: gui/slick/js/core.js [hunk 1]
```diff
@@ -453,24 +466,16 @@ var SICKRAGE = {
...
-                    if (!emby.host) {
-                        $('#emby_host').addClass('warning');
-                    } else {
-                        $('#emby_host').removeClass('warning');
-                    }
-                    if (!emby.apikey) {
-                        $('#emby_apikey').addClass('warning');
-                    } else {
-                        $('#emby_apikey').removeClass('warning');
-                    }
```
Comment (OmgImAlexis) [inline gui/slick/js/core.js]: All of these can now be changed to use the jQuery function I added.

````
 if (!emby.host) {
     $('#emby_host').addClass('warning');
 } else {		
    $('#emby_host').removeClass('warning');		
 }
````

````
$('#emby_host').addRemoveWarningClass(emby.host);
````

**Comment by medariox** [comment]
So I have been testing this for a few hours, trying out everything I could etc. My JS knowledge is fairly limited, but everything seems to be running just fine so far.

What I noticed:
- My linter tells me that https://github.com/pymedusa/SickRage/blob/7dba22759fd4baaf30c13cabbc28fcf28a33f263/gui/slick/js/core.js#L3433 was used before it was defined. Isn't this an issue?
- There are some variables that are not camel case, they will need ```// eslint-disable-line camelcase``` to not fail the tests.

**Comment by OmgImAlexis** [comment]
The issue about the wizard is fine since it's defined in the `vender.min.js` file.

It's already passed travis so that's fine for now. The next PR I'm pushing has most of the other fixes but I didn't want to add onto an already extemely large PR.

**Comment by medariox** [comment]
Approved

Sounds good!

**Comment by fernandog** [comment]
@OmgImAlexis concluded?

**Pull request merged**